[Services] Implement 'Settings' section

carma12 opened the pull request:
The current solution implements the 'Services' > 'Settings' sections excepting `Allowed to retrieve keytab` and `Allowed to create keytab` (those will be made in different PRs).

This PR depends on this one to be merged: https://github.com/freeipa/freeipa-webui/pull/434

mreynolds389 commented:
Adding a user to the keytab sections failed (no such entry).

Authentication Indicators can not be set (error "Unknown option dn")

In fact I can not set anything without getting some error.  Maybe a rebase went wrong?

mreynolds389 commented on src/components/ServicesSections/Provisioning.tsx:
```diff
@@ -5,26 +5,55 @@ import { Flex, FlexItem, Form, FormGroup, Icon } from "@patternfly/react-core";
+}
+
+const Provisioning = (props: PropsToServiceProvisioning) => {
+  console.log("Service: ", props.service);
```
Is this console logging for debugging, can it be removed?  Just curious

carma12 replied:
Oops, leftover.

carma12 commented:
> Adding a user to the keytab sections failed (no such entry).
> 
> Authentication Indicators can not be set (error "Unknown option dn")
> 
> In fact I can not set anything without getting some error. Maybe a rebase went wrong?

As specified in the PR description, the keytab section is the only part that are not implemented in this solution. A lot of code already in the PR :) 

I will check the other error messages and try to fix it. It seems to be what you mentioned...

carma12 commented:
> > Adding a user to the keytab sections failed (no such entry).
> > Authentication Indicators can not be set (error "Unknown option dn")
> > In fact I can not set anything without getting some error. Maybe a rebase went wrong?
> 
> As specified in the PR description, the keytab section is the only part that are not implemented in this solution. A lot of code already in the PR :)
> 
> I will check the other error messages and try to fix it. It seems to be what you mentioned...

The problem has been fixed and ready to review.

mreynolds389 commented:
@carma12 there are still issues

- "Trusted for delegation" checkbox does not retain value after saving (refresh does not help)
- "Trusted for authentication as user" as does not work
- "Requires pre-authentication" gives errors about "no modifications" when trying to save

Besides keytabs (as expected) everything else seems to work

carma12 commented:
> @carma12 there are still issues
> 
>     * "Trusted for delegation" checkbox does not retain value after saving (refresh does not help)
> 
>     * "Trusted for authentication as user" as does not work
> 
>     * "Requires pre-authentication" gives errors about "no modifications" when trying to save
> 
> 
> Besides keytabs (as expected) everything else seems to work

Indeed, something is happening with those checkboxes. Will check...

carma12 commented:
@mreynolds389 - The previous bug has been fixed. Apparently there was a problem with the `IpaCheckbox` component and the `true`|`false` value was not being properly printed, so it should be fine now. The new change looks like this:
```ts
const checked =
    value &&
    ((typeof value === "string" && value.toLowerCase() === "true") ||
      value === true)
      ? true
      : false;
```

mreynolds389 commented:
This fixes the keytabs in the service settings

```
diff --git a/src/components/tables/KeytabTable.tsx b/src/components/tables/KeytabTable.tsx
index e54c06c..e7b88ac 100644
--- a/src/components/tables/KeytabTable.tsx
+++ b/src/components/tables/KeytabTable.tsx
@@ -252,7 +252,7 @@ const KeytabTable = (props: PropsToTable) => {
     let id = props.id;
     if (props.from === "service") {
       // Need to strip off domain for services
-      id = props.id[0].split("@")[0];
+      id = props.id.split("@")[0];
     }
     executeUpdate({
       id: id,
```

carma12 commented:
> This fixes the keytabs in the service settings
> 
> ```
> diff --git a/src/components/tables/KeytabTable.tsx b/src/components/tables/KeytabTable.tsx
> index e54c06c..e7b88ac 100644
> --- a/src/components/tables/KeytabTable.tsx
> +++ b/src/components/tables/KeytabTable.tsx
> @@ -252,7 +252,7 @@ const KeytabTable = (props: PropsToTable) => {
>      let id = props.id;
>      if (props.from === "service") {
>        // Need to strip off domain for services
> -      id = props.id[0].split("@")[0];
> +      id = props.id.split("@")[0];
>      }
>      executeUpdate({
>        id: id,
> ```

Thank you, the changes have been already adapted tot he code.

mreynolds389 reviewed (approved):
LGTM!
